### ISSUE TITLE
Squaring a nonnegative variable

### ISSUE THREAD

**Issue by alexrwatson** [cvxgrp/CVXR#145]
**Describe the bug** 
Introducing the square `x^2` of a nonnegative variable `x` into the objective causes an error when solving using OSQP.

Variations on objective:
- `power(x,2)` and `square(x)` have the same problems.
- `sum_squares(x)` works as expected.

Variations on variable declaration:
- `nonpos=TRUE` has the same problem
- Removing the `nonneg=TRUE` argument does not present the issue
- `PSD=TRUE` does not present the issue
- When `x` is replaced with a vector or matrix the problem persists.

Solver variations:
- `solver="CLARABEL"` and `solver="SCS"` do not present the problem

**To Reproduce** 
```
library(CVXR)
x = Variable(nonneg=TRUE)
result = solve(Problem(Minimize(x^2), list()))
```

**Expected behavior**
Find the optimal solution 0.

**Output**
```
Error in Power(new("Variable", dim = c(1L, 1L), name = "varNA", value = NA_real_,  : 
  argument "p" is missing, with no default
22. .local(.Object, ...)
21. initialize(value, ...)
20. initialize(value, ...)
19. new(structure("Power", package = "CVXR"), ...)
18. .Power(x = x, p = p, max_denom = max_denom) at elementwise.R#813
17. Power(new("Variable", dim = c(1L, 1L), name = "varNA", value = NA_real_,
.is_vector = FALSE, nonneg = FALSE, nonpos = FALSE, complex = FALSE,
imag = FALSE, symmetric = FALSE, diag = FALSE, PSD = FALSE,
NSD = FALSE, hermitian = FALSE, boolean = FALSE, integer = FALSE, ...
16. do.call(class(object), args) at elementwise.R#987
15. copy(object, args = new_args, id_objects = id_objects) at generics.R#1200
14. copy(object, args = new_args, id_objects = id_objects) at canonical.R#64
13. tree_copy(arg, id_objects) at generics.R#1201
12. tree_copy(arg, id_objects) at canonical.R#58
11. tree_copy(problem@objective, id_objects = id2new_obj) at generics.R#1201
10. tree_copy(problem@objective, id_objects = id2new_obj) at reductions.R#366
9. perform(r, problem) at generics.R#1149
8. perform(r, problem) at reductions.R#251
7. perform(object@.intermediate_chain, object) at generics.R#1149
6. perform(object@.intermediate_chain, object) at problem.R#653
5. .construct_chains(object, solver = solver, gp = gp) at problem.R#671
4. psolve(a, solver = NA, ...) at generics.R#966
3. psolve(a, solver = NA, ...) at problem.R#693
2. solve(Problem(Minimize(x^2), list()))
1. solve(Problem(Minimize(x^2), list()))
```

**Version**
```
> sessionInfo()
R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C               LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8    LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/London
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] CVXR_1.0-14

loaded via a namespace (and not attached):
 [1] bit_4.5.0         compiler_4.4.1    R6_2.5.1          Matrix_1.6-5      tools_4.4.1       gmp_0.7-5         rstudioapi_0.17.1
 [8] yaml_2.3.10       Rcpp_1.0.13-1     bit64_4.5.2       grid_4.4.1        renv_1.0.11       Rmpfr_0.9-5       lattice_0.22-5
```

**Comment by bnaras** [comment]
@alexrwatson Thanks for the bug report.  Somehow this bug never got exercised and root cause was a bad call to the `Power()` constructor during a copy.  This is now fixed in version 1.0-15 on the [clarabel_interface](https://github.com/cvxgrp/CVXR/tree/clarabel_interface) branch.

**Comment by alexrwatson** [comment]
@bnaras Thank you, this all seems to be working as expected now.
